src/python/system/path_patcher.py

```diff
@@ -48,9 +48,6 @@ def _is_windows():
 
 def _short_name_modifier(original_path):
   """Get the short path of `path` on windows."""
-  if not _is_windows():
```

Comment:
This is redundant for sure, and unneccessary before every operation. we already don't patch()/unpatch() if we are not on windows.